src/Microsoft.Build.Tasks.Git/GitDataReader/GitRepository.cs

```diff
@@ -209,6 +209,11 @@ public static GitRepository OpenRepository(GitRepositoryLocation location, GitEn
         /// <returns>Null if the HEAD tip reference can't be resolved.</returns>
         internal string? ReadSubmoduleHeadCommitSha(string submoduleWorkingDirectoryFullPath)
         {
+            if (Directory.Exists(Path.Combine(submoduleWorkingDirectoryFullPath, ".git")))
+            {
+                return File.ReadAllText(Path.Combine(submoduleWorkingDirectoryFullPath, ".git", "HEAD"));
+            }
```

Comment:
@crummel can you give some context on this patch? My understanding has been that a submodule by definition has a `.git` file that points to the superproject's Git data for that submodule. Maybe something's being treated as a submodule that shouldn't be?

Reply:
This patch seems to be breaking tests around this scenario, which makes sense: https://github.com/dotnet/sourcelink/blob/8a3edd1902dbfe3adba65f22e3bb7aa2cc73e97f/src/Microsoft.Build.Tasks.Git.UnitTests/GitRepositoryTests.cs#L267-L269

I'm going to revert it for now.

Reply:
The idea here was to use the [Minimal git repository metadata](https://github.com/dotnet/sourcelink/blob/master/docs/README.md) support.  Source-build doesn't have real repos (we purposely use non-git source directories to reduce confusion and tarball size), so we write out .git/HEAD to these folders to make sourcelink work.  This didn't work in submodules because sourcelink always treats submodules as having .gitdir redirects instead of real .git folders (which in a normal situation is true).  I'll take a look and try to figure out a clean fix.

Reply:
Opened https://github.com/dotnet/sourcelink/pull/653 for a better version of this patch.